7a_temp_coop_munge/src/data_parsers/parse_Iowa_DNR_LimnoProfiles_2000_2020.R

```diff
@@ -0,0 +1,507 @@
+#' @param keep_cols chr, vector of column names that should be preserved
+#' @param datetime logical, is there a datetime column?
+#' @param data_start num, row number where the data starts
+#'
+read_iowa_htm_tbl <- function(x, keep_cols, datetime = F, data_start = 3){
+
+  # read data
+  out <- rvest::html_table(rvest::read_html(x, col_names = F)) %>%
+    .[[1]] %>%
+    na_if("") %>% #%>% # scraping HTML often results in `""` in cells at the end
+    remove_na_rows
+
+  # parse and reformat column names
+  names(out) <- out[1, ] %>%
+    unlist %>%
+    tolower # assign col names
+
+  # select columns of interest
+  out <- out %>% dplyr::select(all_of(tolower(keep_cols)))
+
+  # handle date/datetime
+  if(datetime) {
+    out <- out[c(data_start:nrow(out)), ]
+    out[c(2:ncol(out))] <- lapply(out[c(2:ncol(out))], as.numeric)
+
+    # most of the 2011-2016 data contains datetime in `Sampling_Date_Time`
+    # but the time component doesn't always come along for the ride
+    out[, 1] <- tryCatch(lubridate::mdy_hm(unlist(out[ , 1])),
+                         warning = function(w) {lubridate::mdy(unlist(out[ , 1]))})
+  } else {
+    out <- out[c(data_start:nrow(out)), ]
+    out[c(3:ncol(out))] <- lapply(out[c(3:ncol(out))], as.numeric)
+
+    # 2010 data includes a date field and a time field
+    # The final `sapply` removes a few lagging colons (e.g., "9:58:")
+    out[ , 1] <- lubridate::mdy(unlist(out[ , 1]))
+    out[ , 2] <- str_sub(unlist(out[ , 2]), start = 1, end = 5)
+    out[ , 2] <- sapply(unlist(out[ ,2]), function(x) {
+      ifelse(stringr::str_ends(x, ":"),
+             stringr::str_sub(x, 1, 4), x)
+    })
+  }
+
+  return(out)
+
+}
+
+#' Remove rows that are blank (all cells == NA)
+#'
+#' @param x data.frame
+#'
+remove_na_rows <- function(x) {
+  idx <- apply(x, 1, function(x) all(is.na(x)))
+  x[!(apply(x, 1, function(df) all(is.na(df)))), ]
+}
+
+#' Check the maximum depth of the lake profile
+#'
+#' The `explain` file for 2000-2003 Iowa DNR data (`Iowa_profiles_2000-2003.docx`)
+#' notes that some ascent data is included in the datasets, but shouldn't be
+#' used. This function IDs the lowest depth by returning `TRUE` if the maximum
+#' depth is not the final value in the profile
+#'
+#' @param depth_vector num, a vector of depth values
+#'
+check_final_depth <- function(depth_vector) {
+  depth_max <- max(depth_vector, na.rm = T)
+  depth_max_location <- which(depth_vector == depth_max) %>% max
+
+  depth_max_location < length(depth_vector)
+}
+
+#' Trim the lake profile to the maximum depth
+#'
+#' The `explain` file for 2000-2003 Iowa DNR data (`Iowa_profiles_2000-2003.docx`)
+#' notes that some datasets contain data from the descent _and_ the ascent of the data
+#' logger. The ascent data should be trimmed.
+#'
+#' @param df data.frame that includes the depth
+#' @param depth_col int, index for depth column. Defaults to the first column (`depth_col = 1`).
+#'
+trim_to_max_depth <- function(df, depth_col = 1) {
```

Comment:
I don't think we should do this within the parsers. I think separately we want to compare reported depth against profile depths, but I don't think we should screen within the functions. Sometimes the reported depths are wrong and sometimes the water levels fluctuate.

Reply:
In a brief side bar conversation, Jordan and I decided to keep the `trim_to_max_depth` for two reasons:
1. Only a few of the years have this problem, and 
2. IADNR mentions that we should do this in several docs because data includes values from the ascent of the probe (e.g., [2000-2003](https://docs.google.com/document/d/1hhBk6Oz35XRR03jfRvBCXI9XTkLd3yYV/edit) and [2004-2007](https://docs.google.com/document/d/1nr_ODU9IQ_7ns6o960MLekRx84M-UBbi/edit#heading=h.gjdgxs)). These values did not equilibrate.